### PR TITLE
player: add volume and mute notifier

### PR DESCRIPTION
**Pull request by codomania**
Please let me know what you think about this patch for adding volume and mute notifier. If I am on right path then i can work on adding the code in gtk-play to verify volume and mute callbacks.

### CODE REVIEW

File: gtk/gtk-play.c [hunk 1]
```diff
@@ -1430,6 +1431,33 @@ media_info_updated_cb (GstPlayer * player, GstPlayerMediaInfo * media_info,
...
+        volume_changed_cb, play);
+  }
+
+  return FALSE;
+}
+
+static void
+player_volume_changed_cb (GstPlayer * player, GtkPlay * play)
+{
+  g_idle_add ((GSourceFunc) volume_notify_idle_cb, play);
```
Comment (sdroege) [inline gtk/gtk-play.c]: We should dispatch to the main context inside GstPlayer for this, not let the application do it. Just like it's done for all the other signals :)

File: lib/gst/player/gstplayer.c [hunk 2]
```diff
@@ -328,6 +330,16 @@ gst_player_class_init (GstPlayerClass * klass)
       g_signal_new ("media-info-updated", G_TYPE_FROM_CLASS (klass),
       G_SIGNAL_RUN_LAST | G_SIGNAL_NO_RECURSE | G_SIGNAL_NO_HOOKS, 0, NULL,
       NULL, NULL, G_TYPE_NONE, 1, GST_TYPE_PLAYER_MEDIA_INFO);
+
+  signals[SIGNAL_VOLUME_CHANGED] =
+      g_signal_new ("volume-changed", G_TYPE_FROM_CLASS (klass),
+      G_SIGNAL_RUN_LAST | G_SIGNAL_NO_RECURSE | G_SIGNAL_NO_HOOKS, 0, NULL,
+      NULL, NULL, G_TYPE_NONE, 0, G_TYPE_INT, G_TYPE_INT);
```
Comment (sdroege) [inline lib/gst/player/gstplayer.c]: You have two int arguments here, but don't use them anywhere else for the signal, also not during signal emission

File: lib/gst/player/gstplayer.c [hunk 3]
```diff
@@ -2118,6 +2170,18 @@ subtitle_tags_changed_cb (GstElement * playbin, gint stream_index,
       GST_TYPE_PLAYER_SUBTITLE_INFO);
 }
 
+static void
+volume_notify_cb (GObject *obj, GParamSpec *pspec, GstPlayer *self)
+{
+  emit_volume_changed (self);
```
Comment (sdroege) [inline lib/gst/player/gstplayer.c]: Just inline those functions here, you're only ever going to call them from one place anyway :)

**Comment by sdroege** [comment]
Looks generally good, just some minor comments. Thanks for the patch :)

File: gtk/gtk-play.c (hunk 1 above)
Comment (codomania) [inline gtk/gtk-play.c]: Sure, I can takeout the idle loop and do the slider update inside gstplayer context.

File: lib/gst/player/gstplayer.c (hunk 2 above)
Comment (codomania) [inline lib/gst/player/gstplayer.c]: initially i though that our signal with contain old and new volume values but then decided that userapp can query those directly via gstplayer apis and simplify the logic :)

File: gtk/gtk-play.c (hunk 1 above)
Comment (sdroege) [inline gtk/gtk-play.c]: The loop is the same anyway as the GTK app uses dispatch-to-main-context=TRUE on GstPlayer :)

File: lib/gst/player/gstplayer.c (hunk 2 above)
Comment (sdroege) [inline lib/gst/player/gstplayer.c]: Makes sense, so just remove those here and then it seems good :)

**Comment by sdroege** [comment]
Thanks, merged :)

**Pull request closed without merging**